.github/workflows/test.yaml

```diff
@@ -50,3 +50,7 @@ jobs:
         run: npm run format-check --prefix assets
       - name: Run assets tests
         run: npm test --prefix assets
+      - name: Check app build warnings
+        env:
+          MIX_TARGET: app
+        run: mix compile --warnings-as-errors
```

Comment:
I'm running the check after the npm deps have been installed in case the app build depends (or will depend) on a npm dependency. Is that okay?